app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -736,6 +736,18 @@ public boolean onBackPressed() {
             Log.d(TAG, "onBackPressed() called");
         }
 
+        final int behavior = PlayerHelper.getBackBehavior(activity);
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_CLOSE_FULLSCREEN) {
+            if (!isPlayerAvailable() || player.videoPlayerSelected()) {
+                bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+                requireView().postDelayed(this::rotateScreenOnPhoneIfInLandscape, 500);
```

Comment:
This looks suspicious, how did you calculate the 500ms delay to make sure the rotation happens correctly?

Reply:
Oh, forgot to made something about that. This delay was just brute-forced. If it will not work for someone I'll try to find a better way.